src/ameedf/jb8223/assignments/hw02/ClockBonus2.java

```diff
@@ -1,7 +1,138 @@
-package ameedf.jb8223.assignments.hw02;
+package src.ameedf.jb8223.assignments.hw02;
 
 /**
  * The same clock where minimum memory is occupied
  */
 public class ClockBonus2 {
+
+    private TimeUnit timeUnit;
```

Comment:
Although you reduced the memory of **this** class but the total memory has been increased actually (an  extra reference to a TimeUnit instance was added, which in turn has the same variables as before).
HINT for a more efficient way: 
1 second = 1000 milliseconds
1 minutes = 60 seconds = 60 * 1000 milliseconds
1 hour = .....

Reply:
Yes, I've seen Alex's solution and got the idea. Every time unit can be expressed by millisecs.
Oh well... :-)